neuralforecast/common/_base_model.py

```diff
@@ -385,45 +363,65 @@ def on_fit_start(self):
         random.seed(self.random_seed)
 
     def configure_optimizers(self):
-        if self.optimizer:
-            optimizer_signature = inspect.signature(self.optimizer)
-            optimizer_kwargs = deepcopy(self.optimizer_kwargs)
-            if "lr" in optimizer_signature.parameters:
-                if "lr" in optimizer_kwargs:
-                    warnings.warn(
-                        "ignoring learning rate passed in optimizer_kwargs, using the model's learning rate"
-                    )
-                optimizer_kwargs["lr"] = self.learning_rate
-            optimizer = self.optimizer(params=self.parameters(), **optimizer_kwargs)
-        else:
-            if self.optimizer_kwargs:
-                warnings.warn(
-                    "ignoring optimizer_kwargs as the optimizer is not specified"
+        if self.config_optimizers is not None:
+            # return the customized optimizer settings if specified
+            return self.config_optimizers
+
+        # default choice
+        optimizer = torch.optim.Adam(self.parameters(), lr=self.learning_rate)
+        scheduler = {
+            "scheduler": torch.optim.lr_scheduler.StepLR(
+                optimizer=optimizer, step_size=self.lr_decay_steps, gamma=0.5
+            ),
+            "frequency": 1,
+            "interval": "step",
+        }
+        return {"optimizer": optimizer, "lr_scheduler": scheduler}
+
+    def set_configure_optimizers(
```

Comment:
I was thinking of taking an argument in the constructor that has a function with the same signature as `configure_optimizers` and modify the method to be something like:

```python
def configure_optimizers(self):
    if self.config_optimizers is not None:
        return self.config_optimizers(self)
```
to reduce the friction for users, since they may already know how to define that method and we just want to let them kind of override it. WDYT?

Reply:
I think it needs self for the model parameters at least, not sure if other hyperparameters as well